src/schema/nmdc.yaml

```diff
@@ -686,6 +704,12 @@ slots:
     range: FluidHandling
     description: how a processed sample is introduced into a mass spectrometer, for example liquid chromatography or direct infusion through a syringe
 
+  has_configuration:
+    any_of:
```

Comment:
Why would the range be `Configuration` in some cases and `DataObject` in other cases?

Reply:
We should go to great lengths to avoid `any_of` `ranges`. They are complex to reason over and diagram. That may change in the future.

Reply:
@brynnz22  and I were thinking ahead to accommodate workflow configuration files (`DataObjects`) that will be associated with a `has_configuration` slot on `WorkflowExcecution` class.  See https://github.com/microbiomedata/nmdc-schema/issues/1912 .

Reply:
I would like to make sure it's **very** clear we we
- embed configuration-like data in an instance of a `PlannedProcess` 
- associate configuration-like data with a process via a `DataObject`, which is a pointer to a configuration file (but doesn't actually manifest the settings in the instance)
- embed configuration-like data in instances of the new `Configuration` class

Reply:
A review of those configuration embedding/linking options may shine some additional light on the idea of a configurations as platforms

Reply:
Sorry I do not quite understand this.

> I would like to make sure it's **very** clear we we
> 
> * embed configuration-like data in an instance of a `PlannedProcess`
> * associate configuration-like data with a process via a `DataObject`, which is a pointer to a configuration file (but doesn't actually manifest the settings in the instance)
> * embed configuration-like data in instances of the new `Configuration` class

Are there actions we can take in the PR to meet this?

Reply:
Since this PR adds a new way of capturing configuration, it is required that this PR includes documentation (or additional schema constraints) on  on how it relates to existing ways of capturing configuration information.